### PR TITLE
picard-tools: import from homebrew/science

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Migrate from homebrew/science.

### CODE REVIEW

File: Formula/picard-tools.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  def caveats
+    <<-EOS.undent
+      The Picard JAR files are installed to
+        #{HOMEBREW_PREFIX}/share/java
+    EOS
+  end
+
+  test do
+    assert_match "USAGE", shell_output("java -jar #{share}/java/picard.jar -h 2>&1", 1)
```
Comment (DomT4) [inline Formula/picard-tools.rb]: Is this not made redundant from the line below?

Comment (sjackman) [reply Formula/picard-tools.rb]: Deleted.

File: Formula/picard-tools.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class PicardTools < Formula
+  desc "Tools for manipulating HTS data and formats"
+  homepage "https://broadinstitute.github.io/picard/"
+  url "https://github.com/broadinstitute/picard/releases/download/2.16.0/picard.jar"
+  sha256 "01cf3c930d2b4841960497491512d327bf669c1ed2e753152e1e651a27288c2d"
+
+  bottle :unneeded
+
+  depends_on :java
```
Comment (fxcoudert) [inline Formula/picard-tools.rb]: The website says _“For the tools to run properly, you must have Java 1.8 installed”_. This needs to be reflected here.

File: Formula/picard-tools.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  def install
+    java = share/"java"
+    java.install Dir["*.jar"]
+    bin.write_jar_script java/"picard.jar", "picard"
+  end
+
+  test do
+    assert_match "USAGE", shell_output("#{bin}/picard -h 2>&1", 1)
```
Comment (fxcoudert) [inline Formula/picard-tools.rb]: We need a test that does more than output usage info

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/picard-tools.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/picard-tools.rb]: Do we know for a fact that it doesn't work with Java 9?

Comment (fxcoudert) [reply Formula/picard-tools.rb]: It should have at least 1.8+

File: Formula/picard-tools.rb (hunk 3 above)
Comment (sjackman) [reply Formula/picard-tools.rb]: I've added a test.

**Review by fxcoudert — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/picard-tools.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    Pathname.new("test.fasta").write <<~EOS
```
Comment (DomT4) [inline Formula/picard-tools.rb]: `(testpath/"test.fasta").write <<~EOS`.

**Comment by sjackman** [comment]
Is this the preferred commit log message?
```
picard-tools 2.15.0 (new formula)

Migrate from homebrew/science.
```

File: Formula/picard-tools.rb (hunk 4 above)
Comment (sjackman) [reply Formula/picard-tools.rb]: Done.

**Comment by fxcoudert** [comment]
`picard-tools: import from homebrew/science` (as in the title) is the preferred commit message

**Comment by sjackman** [comment]
Shall I include the version number?
```
picard-tools 2.15.0: import from homebrew/science
```

File: Formula/picard-tools.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  homepage "https://broadinstitute.github.io/picard/"
+  url "https://github.com/broadinstitute/picard/releases/download/2.16.0/picard.jar"
+  sha256 "01cf3c930d2b4841960497491512d327bf669c1ed2e753152e1e651a27288c2d"
+
+  bottle :unneeded
+
+  depends_on :java => "1.8+"
+
+  def install
+    java = share/"java"
```
Comment (ilovezfs) [inline Formula/picard-tools.rb]: we use libexec

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/picard-tools.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  test do
+    assert_match "USAGE", shell_output("#{bin}/picard -h 2>&1", 1)
+    (testpath/"test.fasta").write <<~EOS
+      >U00096.2:1-70
+      AGCTTTTCATTCTGACTGCAACGGGCAATATGTCT
+      CTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGC
+    EOS
+    assert_match "TCTCTG", shell_output("#{bin}/picard NormalizeFasta I=test.fasta O=/dev/stdout")
```
Comment (ilovezfs) [inline Formula/picard-tools.rb]:
```
    cmd = "#{bin}/picard NormalizeFasta I=test.fasta O=/dev/stdout"
    assert_match "TCTCTG", shell_output(cmd)
```

File: Formula/picard-tools.rb (hunk 5 above)
Comment (sjackman) [reply Formula/picard-tools.rb]: Fixed.

File: Formula/picard-tools.rb (hunk 6 above)
Comment (sjackman) [reply Formula/picard-tools.rb]: Done.

**Review by ilovezfs — APPROVED** [review]

**Comment by sjackman** [comment]
Thanks for merging, @fxcoudert !

**Pull request merged**